src/features/Courses/CourseDetailTable/columns.jsx

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable react/prop-types */
+import React, { useState } from 'react';
+import { Badge, Button } from 'react-paragon-topaz';
+import { useDispatch, useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { format } from 'date-fns';
+
+import { fetchClassesData } from 'features/Classes/data/thunks';
+import { updateClassSelected } from 'features/Instructors/data/slice';
+import AssignInstructors from 'features/Instructors/AssignInstructors';
+
+const columns = [
+  {
+    Header: 'Class',
+    accessor: 'className',
+    Cell: ({ row }) => (<span className="text-truncate">{row.values.className}</span>),
+  },
+  {
+    Header: 'Instructor',
+    accessor: 'instructors',
+    Cell: ({ row }) => {
+      const { classId } = useParams();
+      const [isModalOpen, setIsModalOpen] = useState(false);
+
+      const dispatch = useDispatch();
+      const institution = useSelector((state) => state.main.selectedInstitution);
+
+      const handleOpenModal = () => {
+        dispatch(updateClassSelected(row.original.classId));
+        setIsModalOpen(true);
+      };
+
+      const handleCloseModal = () => {
+        dispatch(fetchClassesData(institution.id, 1, classId));
+        setIsModalOpen(false);
+      };
+
+      if (row.values.instructors?.length > 0) {
+        return (
+          <ul className="instructors-list mb-0">
+            {row.values.instructors.map(instructor => <li key={instructor} className="text-truncate">{`${instructor}`}</li>)}
+          </ul>
+        );
+      }
+
+      return (
+        <>
+          <Button onClick={handleOpenModal} className="button">
```

Comment:
Styles doesn't match with the wireframe: https://www.figma.com/file/pRMTr6L9S9oNFV3qZgyHcn/Achieve-Dashboard---11.02.2023?type=design&node-id=1506%3A25667&mode=design&t=sTSslPjEGS03ghM0-1
Plese use small version